### PR TITLE
📦 NEW: Add build tools (#8)

### LINKED ISSUE #8: Add build tools
See woocommerce/storefront-blog-customiser#25

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #8

Add build tools to generate CSS and translations files.

### Screenshots

n/a

### How to test the changes in this Pull Request:

1. Check out this PR
2. Run `npm i` to install all dependencies → Dependencies should be installed
3. Run `npm run build` to build the extension → The file `storefront-footer-bar.zip` should be created in root
4. Run `npm run build:dev` to render JS and SCSS files → The files `assets/css/style.css`, `assets/css/style.css.map`, `assets/css/style.min.css`, `assets/js/customizer.js`, `assets/js/customizer.min.js` & `assets/js/customizer.min.js.map` will be rendered
5. Run `npm run start` to watch SCSS changes → If `src/css/style.scss` will be changes, the files `assets/css/style.css`, `assets/css/style.css.map` & `assets/css/style.min.css` will be rendered
6. Run `npm run watch` to watch SCSS changes → If `src/css/style.scss` will be changes, the files `assets/css/style.css`, `assets/css/style.css.map` & `assets/css/style.min.css` will be rendered
7. Run `npm run js` to render JS files →  The files `assets/js/customizer.js`, `assets/js/customizer.min.js` & `assets/js/customizer.min.js.map` will be rendered
8. Run `npm run css` to render SCSS files → The files `assets/css/style.css`, `assets/css/style.css.map` & `assets/css/style.min.css` will be rendered
9. Run `npm run makepot` to render the POT file → The file `languages/storefront-footer-bar.pot` will be rendered

### Changelog

> Dev - Add build tools to generate CSS and translations files. #8

### Tests

n/a

### Note

Due to insufficient permissions, I wasn't able to send this PR against this repo. Therefore, I had to send the PR via my forked repo.

cc: @jconroy (Pinging you here as I'm also unable to request a review from you.)

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+					require: 'susy',
+					implementation: sass,
+					sourceMap: true,
+					includePaths: ['node_modules/susy/sass'].concat( require( 'node-bourbon' ).includePaths )
+				},
+				files: [
+					{
+						expand: true,
+						cwd: `${sourceFolderRelPath}/scss`,
+						src: ['**/*.scss¿'],
```
Comment (findingsimple) [inline Gruntfile.js]: Thanks @nielslange !

I noticed the CSS map files weren't generated and then can see an unusual ¿ character here, wondering if that is the cause?

File: languages/storefront-footer-bar.pot [hunk 2]
```diff
@@ -1,51 +1,52 @@
-# Copyright (C) 2015 Storefront Footer Bar
+# Copyright (C) 2020 WooThemes
```
Comment (findingsimple) [inline languages/storefront-footer-bar.pot]: Lets update the plugin header info while we are at

**Comment by findingsimple** [comment]
Thanks @nielslange just add a few comments for things I noticed. The only extra thing I've done with others is add a readme.md

Update - ignore my comment about a readme, you have create an issue for that I didn't realize was there https://github.com/woocommerce/storefront-footer-bar/issues/11

File: Gruntfile.js (hunk 1 above)
Comment (nielslange) [reply Gruntfile.js]: > I noticed the CSS map files weren't generated and then can see an unusual ¿ character here, wondering if that is the cause?

Yes, that causes the problem. The ¿ character should not be there. I'll update this PR accordingly and will also adjust the plugin header as requested in https://github.com/woocommerce/storefront-footer-bar/pull/12#discussion_r495521920.

Comment (nielslange) [reply Gruntfile.js]: Fixed 😀

File: languages/storefront-footer-bar.pot (hunk 2 above)
Comment (nielslange) [reply languages/storefront-footer-bar.pot]: Fixed 😀

**Comment by nielslange** [comment]
> Thanks @nielslange just add a few comments for things I noticed. The only extra thing I've done with others is add a readme.md
> 
> Update - ignore my comment about a readme, you have create an issue for that I didn't realize was there #11

All changes and I might add the missing README.md files to all SF extensions later today. As adding them is a change that is not noticeable by users of the extensions, but only by developers, I'd merge the READEME.md PRs directly.

**Pull request closed without merging**